graphs/possible_bipartition.py

```diff
@@ -8,5 +8,39 @@ def possible_bipartition(dislikes):
         Time Complexity: ?
         Space Complexity: ?
     """
-    pass
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
+    
+    if not dislikes:
+        return True
+    
+    adj_list = {}
+    for i in range(len(dislikes)):
+        adj_list[i] = dislikes[i]
+                
+    colors = {}
+    for pair in dislikes:
+        for num in pair:
+            if num not in colors:
+                colors[num] = -1
+
+    for i in range(len(dislikes)):
+        if dislikes[i]:
+            break
+    
+    src = dislikes[i][0]
```

Comment:
🤔 Generally not ideal code style to refer to a variable, `i` outside of its scope.

```suggestion
    for i in range(len(dislikes)):
        if dislikes[i]:
            src = dislikes[i][0]
            break
```